src/core/image_resizer.js

```diff
@@ -96,6 +96,30 @@ class ImageResizer {
     return area > maxArea;
   }
 
+  static getReducePowerForJPX(width, height, componentsCount) {
+    const area = width * height;
+    // The maximum memory we've in the wasm runtime is 2GB.
+    // Each component is 4 bytes and we can't allocate all the memory just for
+    // the buffers so we limit the size to 1GB / (componentsCount * 4).
+    // We could use more than 2GB by setting MAXIMUM_MEMORY but it would take
+    // too much time to decode a big image.
+    const maxJPXArea = 2 ** 30 / (componentsCount * 4);
+    if (!this.needsToBeResized(width, height)) {
+      if (area > maxJPXArea) {
+        // The image is too large, we need to rescale it.
+        return Math.ceil(Math.log2(area / maxJPXArea));
+      }
+      return 0;
+    }
+    const { MAX_DIM, MAX_AREA } = this;
+    const minFactor = Math.max(
+      width / MAX_DIM,
+      height / MAX_DIM,
+      Math.sqrt((width * height) / Math.min(maxJPXArea, MAX_AREA))
```

Comment:
Can we use the local variable?
```suggestion
      Math.sqrt(area / Math.min(maxJPXArea, MAX_AREA))
```